client/src/app/components/sidebar/object-inpsector/object-insepctor.component.ts

```diff
@@ -0,0 +1,90 @@
+        REACTIVE_FORM_DIRECTIVES,
+        FORM_DIRECTIVES
+    ],
+    template: `
+            <form class="object-inspector-component">
+                <div class="formHead">
+                     <span class="edit-text">Edit</span>
+                    <i class="fa fa-info-circle info-icon"></i>
+                </div>
+            
+                <div class="form-group">
+                    <label for="inputId">ID</label>
+                    <input type="text" 
+                           name="inputId" 
+                           class="form-control"
+                           [(ngModel)]="selectedProperty.id">
+                </div>
+                
+                <div class="form-group">
+                    <label for="inputType">Type</label>
+                    
+                    <select class="form-control" 
+                    name="dataType"
+                    [(ngModel)]="selectedProperty.type" required>
+                        <option *ngFor="let propertyType of propertyTypes" [value]="propertyType">
+                            {{propertyType}}
+                        </option>
+                    </select>
+                </div>
+                
+                <div class="form-group">
+                    <label for="inputValue">Value</label>
+                    <expression-input [inputControl]="objectInspectorForm.controls['expression']"
+                                      [(expression)]="selectedProperty.value">
+                    </expression-input>
+                </div>
+            </form>
+    `
+})
+export class ObjectInspectorComponent implements DoCheck, OnChanges {
+
+    /** The object that we are editing */
+    @Input()
+    private inputModelStream: BehaviorSubject<InputProperty>;
+
+    /** The currently displayed property */
+    private selectedProperty: InputProperty;
+
+    /** A copy of the property for change detection */
+    private oldProperty: InputProperty;
+
+    private objectInspectorForm: FormGroup;
+
+    /** Possible property types */
+    private propertyTypes: Array<string> = ["File", "string", "enum", "int", "float", "boolean", "array", "record", "map"];
+
+    constructor(private formBuilder: FormBuilder) {
+        this.objectInspectorForm = this.formBuilder.group({
+            expression: ['', Validators.compose([Validators.required, Validators.minLength(1)])]
+        });
+    }
+
+    /** This gets triggered when a data-bound input property value changes, i.e. when we change the inputModelStream. */
+    ngOnChanges(): void {
+        this.inputModelStream.subscribe((inputPort: InputProperty) => {
+            this.selectedProperty = inputPort;
+        });
+    }
+
+    /** Custom change detection needed for objects */
+    ngDoCheck(): void {
+        if (!_.isEqual(this.selectedProperty, this.oldProperty)) {
+            this.oldProperty = _.cloneDeep(this.selectedProperty);
+            this.inputModelStream.next(this.selectedProperty)
+        }
```

Comment:
I use custom change detection because the OnChanges hook will not be triggered if object properties are changed, its triggered only for primitive values and arrays. I think changing `selectedProperty` to be `BehaviorSubject` will work, its good suggestion.